src/analysis/bounded_variation.lean

```diff
@@ -54,7 +53,7 @@ the supremum of the sum of `edist (f (u (i+1))) (f (u i))` over all finite incre
 sequences `u` in `s`. -/
 noncomputable def evariation_on (f : α → E) (s : set α) : ℝ≥0∞ :=
 ⨆ (p : ℕ × {u : ℕ → α // monotone u ∧ ∀ i, u i ∈ s}),
-  ∑ i in finset.range p.1, edist (f ((p.2 : ℕ → α) (i+1))) (f ((p.2 : ℕ → α) i))
+  ∑ i in finset.range p.1, edist (f (p.2.val (i+1))) (f (p.2.val i))
```

Comment:
Don't do that: the simp normal form is the coercion, not `.val`.

Reply:
Coercion seems to take much longer time to elaborate than `subtype.val` in my experience (in this file).

Reply:
Since everything around coercions is going to change in Lean 4, current elaboration times shouldn't really matter here. Doing the right thing so that mathport works fine and translates the file to idiomatic Lean 4 is much more important IMHO.